flink-dist/pom.xml

```diff
@@ -403,15 +403,6 @@ under the License.
 				</configuration>
 			</plugin>
 
-			<!-- binary compatibility checks -->
```

Comment:
I think this is an artifact from the time when the checker was defined in the root pom and we disabled it for all modules that don't need checking.